File: baekjoon/not-classified/1463/1463.py
n = int(input())

d = {}

d[1] = 0
def solution(n):
    if n in d:
        return d[n]

    d[n] = solution(n - 1) + 1

    if (n % 2 == 0):
        d[n] = min(solution(n / 2) + 1, d[n])
    
    if (n % 3 == 0):
        d[n] = min(solution(n / 3) + 1, d[n])

    return d[n]

print(solution(n))

File: baekjoon/not-classified/1463/1463.js
var n = Number(require('fs').readFileSync('/dev/stdin').toString());
d = []
d[1] = 0

var solution = function(n) {
    console.log(n);
    if (d[n] !== undefined) {
        return d[n];
    }

    d[n] = solution(n - 1) + 1;

    if (n % 2 === 0){
        d[n] = Math.min(solution(n / 2) + 1, d[n]);
    }
    if (n % 3 === 0){
        d[n] = Math.min(solution(n / 3) + 1, d[n]);
    }
    return d[n];
}
console.log(solution(n));